System/FilePath/Glob/Utils.hs

```diff
@@ -125,6 +127,10 @@ doesDirectoryExist s =
                else return False
 #endif
 
+#if mingw32_HOST_OS
+foreign import ccall unsafe "windows.h GetFileAttributesW" c_GetFileAttributes :: LPCTSTR -> IO FileAttributeOrFlag
```

Comment:
I think `ccall` is not always correct: https://github.com/haskell/win32/blob/master/include/windows_cconv.h

Reply:
32-bit Windows support of GHC is already dropped at GHC 8.6.2, so I think this glob package can drop support for it.

https://www.haskell.org/ghc/download_ghc_8_6_4.html

Reply:
travis-ci.org is now dead so the CI here no longer works, but at least at last check Glob still compiled on GHC 8.0: https://github.com/Deewiant/glob/blob/090181e60b448a1ce0e18fe5f5783139f420e721/.travis.yml#L2

Of course this was only tested on Linux, but I'd expect the same to hold for Windows. As such this would be moving the minimum supported GHC version on Windows by a number of releases. It's probably fair enough — I'll let you be the judge of that — but I'd say it's worth an explicit call-out in the eventual release notes, at least.

Reply:
I added a piece of code for x86. https://github.com/Deewiant/glob/pull/41/commits/f897d6aab75fcd29bfa2ee455cef4595f2721600